test/checks/design/InterfaceIsTypeCheckTest.hx

```diff
@@ -0,0 +1,40 @@
+package checks.design;
+
+import checkstyle.checks.design.InterfaceIsTypeCheck;
+
+class InterfaceIsTypeCheckTest extends CheckTestCase<InterfaceIsTypeCheckTests> {
+
+	public function testWithJustVars() {
+		assertMsg(new InterfaceIsTypeCheck(), InterfaceIsTypeCheckTests.TEST1, "Interfaces should describe a type and hence have methods");
+	}
+
+	public function testMarkerInterface() {
+		assertNoMsg(new InterfaceIsTypeCheck(), InterfaceIsTypeCheckTests.TEST2);
+	}
+
+	public function testNoMarkerInterface() {
+		var chk = new InterfaceIsTypeCheck();
+		chk.allowMarkerInterfaces = false;
+		assertMsg(chk, InterfaceIsTypeCheckTests.TEST2, "Interfaces should describe a type and hence have methods");
```

Comment:
Then there's the question of what kind of properties though. Just properties with getters / setters? `(default, null)` for example is not too different from a variable. I guess the question here is what problem this check is supposed to solve.